
# 20260321, Adriaan van Os <adriaan@adriaan.biz>
# revert this crazy compiler commit
# https://gitlab.com/freepascal.org/fpc/source/-/commit/2e2f2eb78467fd70d59196033a7197e9428f3992
# that says that "If SysUtils is not included, run-time error should be raised regardless from the try..except block"
# That is insane. Even if it were to be true, then either SysUtils should be included implicitly (like the
# System unit) or the compiler should refuse the try ... except block.

--- a/rtl/inc/system.inc	2026-03-11 14:43:54
+++ b/rtl/inc/system.inc	2026-03-21 10:36:09
@@ -1423,6 +1423,10 @@
   errorcode:=word(Errno);
   erroraddr:=addr;
   errorbase:=frame;
+{$ifdef FPC_HAS_FEATURE_EXCEPTIONS}
+  if ExceptAddrStack <> nil then
+    raise TObject(nil) at addr,frame;
+{$endif FPC_HAS_FEATURE_EXCEPTIONS}
   Halt(errorcode);
 end;
 
